internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -551,78 +577,152 @@ func resourceLinuxVirtualMachineCreate(d *pluginsdk.ResourceData, meta interface
 		Tags: tags.Expand(t),
 	}
 
-	if diskControllerType, ok := d.GetOk("disk_controller_type"); ok {
-		params.Properties.StorageProfile.DiskControllerType = pointer.To(virtualmachines.DiskControllerTypes(diskControllerType.(string)))
+	osDiskRaw := d.Get("os_disk").([]interface{})
+	osDisk, err := expandVirtualMachineOSDisk(osDiskRaw, virtualmachines.OperatingSystemTypesLinux)
+	if err != nil {
+		return fmt.Errorf("expanding `os_disk`: %+v", err)
 	}
 
-	if encryptionAtHostEnabled, ok := d.GetOk("encryption_at_host_enabled"); ok {
-		if encryptionAtHostEnabled.(bool) {
-			if virtualmachines.SecurityEncryptionTypesDiskWithVMGuestState == virtualmachines.SecurityEncryptionTypes(securityEncryptionType) {
-				return fmt.Errorf("`encryption_at_host_enabled` cannot be set to `true` when `os_disk.0.security_encryption_type` is set to `DiskWithVMGuestState`")
+	securityEncryptionType := ""
+
+	if !osDiskIsImported {
+		securityEncryptionType = osDiskRaw[0].(map[string]interface{})["security_encryption_type"].(string)
+		var computerName string
+		if v, ok := d.GetOk("computer_name"); ok && len(v.(string)) > 0 {
+			computerName = v.(string)
+		} else {
+			_, errs := computeValidate.LinuxComputerNameFull(d.Get("name"), "computer_name")
+			if len(errs) > 0 {
+				return fmt.Errorf("unable to assume default computer name %s Please adjust the %q, or specify an explicit %q", errs[0], "name", "computer_name")
```

Comment:
Nit: we generally wrap property names in backticks rather than quotes

```suggestion
				return fmt.Errorf("unable to assume default computer name %s. Please adjust the `name`, or specify an explicit `computer_name`", errs[0])
```

Reply:
Not new, fine to ignore, I see there's many other error messages in this and the windows VM resource that have similar patterns.

Reply:
Copy-paste/refactor error I missed 😞 - We should update for consistency when we touch.